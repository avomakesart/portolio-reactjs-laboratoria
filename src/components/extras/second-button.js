import React from "react";

export function SecondButton({ label }) {
  return (
    <div data-testid="button" className="button-container">
      Hi {label}
    </div>
  );
}
