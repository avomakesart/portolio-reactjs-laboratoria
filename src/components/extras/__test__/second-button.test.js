import React from "react";
import ReactDom from "react-dom";
import { SecondButton } from "../second-button";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";

import renderer from "react-test-renderer";

afterEach(cleanup);

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDom.render(<SecondButton></SecondButton>, div);
});

it("renders correctly", () => {
  const { getByTestId } = render(
    <SecondButton label="click me please"></SecondButton>
  );
  expect(getByTestId("button")).toHaveTextContent("click me please");
});

it("matches snapshot", () => {
  const tree = renderer
    .create(<SecondButton label="click me please"></SecondButton>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
