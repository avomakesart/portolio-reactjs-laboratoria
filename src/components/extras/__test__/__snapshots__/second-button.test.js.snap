// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="button-container"
  data-testid="button"
>
  Hi 
  click me please
</div>
`;
